app/views/activity_notification/subscriptions/default/_subscription.html.erb

```diff
@@ -24,12 +24,12 @@
           <% end %>
         <% else %>
           <% if ActivityNotification.config.subscribe_as_default %>
-            <%= link_to subscribe_path_for(subscription, option_params), onclick: '$(this).find("input").prop("checked", true);$(this).parent().parent().parent().next().slideDown();$(this).parent().parent().parent().next().find("input").prop("checked", true);$(this).parent().parent().parent().next().next().slideDown();$(this).parent().parent().parent().next().next().find("input").prop("checked", true);', method: :put, remote: true do %>
+            <%= link_to subscribe_path_for(subscription, option_params.merge(with_email_subscription: ActivityNotification.config.subscribe_to_email_as_default, with_optional_targets: ActivityNotification.config.subscribe_to_optional_targets_as_default)), onclick: "$(this).find(\"input\").prop(\"checked\", true);$(this).parent().parent().parent().next().slideDown();$(this).parent().parent().parent().next().find(\"input\").prop(\"checked\", #{ActivityNotification.config.subscribe_to_email_as_default.to_s});$(this).parent().parent().parent().next().next().slideDown();$(this).parent().parent().parent().next().next().find(\"input\").prop(\"checked\", #{ActivityNotification.config.subscribe_to_optional_targets_as_default});", method: :put, remote: true do %>
```

Comment:
Why do you set default parameters in a view? It seems to be too complex and it might become breaking changes for the existing users using customized views. I think subscription records should be initialized with default subscription in server side business logic, not front end views. What do you think about it?

Reply:
Nice catch. I think I just haven't found the right place at first. There are two separate changes in this line.

1.

```ruby
option_params.merge(with_email_subscription: ActivityNotification.config.subscribe_to_email_as_default, with_optional_targets: ActivityNotification.config.subscribe_to_optional_targets_as_default)
```
This should be reverted, there's a better place for it in [subscriptions_controller.rb#L93-L97](https://github.com/simukappu/activity_notification/blob/dde51036596ca86552bd6c9d032354e128a7f417/app/controllers/activity_notification/subscriptions_controller.rb#L93-L97), I just need to change the `to_boolean(true)` conversions to use the right default from config.

2.

```javascript
.find("input").prop("checked", #{ActivityNotification.config.subscribe_to_email_as_default.to_s});
```
 
This code should remain here to remove a flick in the transitions. Without this, for a few seconds the frontend could show a wrong state, then it got rewritten with the response from the backend.

I'm going to make a new commit about these.